PocketCastsTests/Tests/IAP/IAPHelperTests.swift

```diff
@@ -0,0 +1,78 @@
+import XCTest
+import StoreKitTest
+
+@testable import podcasts
+import PocketCastsServer
+
+final class IAPHelperTests: XCTestCase {
+
+    override func setUpWithError() throws {
+        // Put setup code here. This method is called before the invocation of each test method in the class.
+    }
+
+    override func tearDownWithError() throws {
+        // Put teardown code here. This method is called after the invocation of each test method in the class.
+    }
+
+    let configurationFile = "Pocket Casts Configuration"
+
+    func testRequestInfo() throws {
+        let session = try SKTestSession(configurationFileNamed: configurationFile)
+        session.clearTransactions()
+        session.resetToDefaultState()
+        session.disableDialogs = true
+
+        let helper = IAPHelper()
+        let expectation = XCTestExpectation(description: "Fetch Products")
+        NotificationCenter.default.addObserver(forName: ServerNotifications.iapProductsUpdated, object: nil, queue: nil) { notification in
+            expectation.fulfill()
+        }
+        helper.requestProductInfo()
+
+        wait(for: [expectation], timeout: 5)
+        XCTAssert(helper.hasLoadedProducts)
+
+        let _ = helper.getProduct(for: .monthly)
+
+        let price = helper.getPrice(for: .monthly)
+        XCTAssertEqual(price, "$3.99")
+
+        let priceWithFrequency = helper.getPriceWithFrequency(for: .monthly)
+        XCTAssertEqual(priceWithFrequency, "$3.99 per month")
+
+        let paymentFrequency = helper.getPaymentFrequency(for: .monthly)
+        XCTAssertEqual(paymentFrequency, "month")
+    }
+
+    func testPurchase() throws {
```

Comment:
This test fails to me when running standalone and on a clean build because `ServerSettings.syncingEmail()` is `nil`. It's probably succeeding in CI because the `EndOfYearStoriesBuilderTests` set it on the `setUp`.

Also, there's a transaction after this test run. Is it possible to delete/clean that?

Reply:
I'm adding code to clean up the session after the test.